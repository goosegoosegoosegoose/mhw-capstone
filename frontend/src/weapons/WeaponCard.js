import React from "react";
import { Card } from "react-bootstrap";
import { useNavigate } from "react-router-dom";
import ToggleButton from "../common/ToggleButton";


const WeaponCard = ({id, name, img, add, remove}) => {
  const nav = useNavigate();

  const handleClick = () => {
    nav(`/weapons/w/${id}`)
  }

  return (
    <Card className="my-2 mx-1 col-sm-2" bg="dark">
      <div className="row justify-content-center">
        <div className="col-sm-8 pt-2 mt-2">
          {img ? <Card.Img onClick={handleClick} variant="top" style={{cursor: "pointer"}} src={img} alt="weapon" /> : <></>}
        </div>
      </div>
      <Card.Body className="d-flex flex-column">
        <Card.Title style={{cursor: "pointer"}} onClick={handleClick}>{name}</Card.Title>
        <ToggleButton id={Number(id)} type="weapons" spacing="my-1 mt-auto" add={add} remove={remove} label1="Sell" label2="Craft"/>
      </Card.Body>
    </Card>
  )
}

export default WeaponCard;