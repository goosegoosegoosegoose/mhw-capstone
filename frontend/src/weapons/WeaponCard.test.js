import { render } from '@testing-library/react';
import WeaponCard from "./WeaponCard";
import { MemoryRouter } from 'react-router-dom';

test('smoke test', () => {
  render(
    <MemoryRouter>
      <WeaponCard />
    </MemoryRouter>
  );
});

test('snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <WeaponCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});