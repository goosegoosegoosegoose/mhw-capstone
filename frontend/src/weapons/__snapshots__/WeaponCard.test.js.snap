// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-2 mx-1 col-sm-2 card bg-dark"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-sm-8 pt-2 mt-2"
      />
    </div>
    <div
      class="d-flex flex-column card-body"
    >
      <div
        class="card-title h5"
        style="cursor: pointer;"
      />
      <button
        class="my-1 mt-auto btn btn-danger btn-sm"
        disabled=""
        type="button"
      >
        Craft
      </button>
    </div>
  </div>
</DocumentFragment>
`;
